### PR TITLE
feat(replays): Add materialized view query entrypoints

### PR DESCRIPTION
**Pull request by cmanallen**
Adds entry points for the materialized view so that when its deployed we can begin using it.

Materialized view in question:
* https://github.com/getsentry/snuba/pull/5708

### CODE REVIEW

**Comment by bruno-garcia** [comment]
The query in the PR would test against this materialized view?
* https://github.com/getsentry/snuba/pull/5708

sorry noob here about this stuff

**Comment by cmanallen** [comment]
@bruno-garcia Yep!

File: src/sentry/replays/usecases/query/__init__.py [hunk 1]
```diff
@@ -265,7 +294,55 @@ def query_using_optimized_search(
...
+            Condition(Column("to_hour_timestamp"), Op.LT, period_stop),
+            Condition(Column("to_hour_timestamp"), Op.GTE, period_start),
+        ],
+        having=having,
+        orderby=orderby,
+        groupby=[Column("replay_id")],
+    )
+
+
+def make_materialized_view_query(
```
Comment (aliu39) [inline src/sentry/replays/usecases/query/__init__.py:330]: don't notice any references to this fx, what was it intended for? @cmanallen

File: src/sentry/replays/usecases/query/__init__.py [hunk 2]
```diff
@@ -212,7 +221,21 @@ def query_using_optimized_search(
...
+    ):
+        query = make_materialized_view_search_query(
+            search_filters=search_filters,
+            sort=sort,
+            project_ids=project_ids,
+            period_start=period_start,
+            period_stop=period_stop,
+        )
+        referrer = "replays.query.browse_materialized_view_conditions_subquery"
+        source = "materialized-view"
```
Comment (aliu39) [inline src/sentry/replays/usecases/query/__init__.py:237]:
```suggestion
        source = "materialized-view-query"
```

**Comment by aliu39** [comment]
Correct me if I'm wrong but we can't test right now since MV doesn't exist in local + CI environments. Would be good to duplicate all tests involving filter and sort fields in the mv config. Can parametrize with https://stackoverflow.com/a/65820397

**Essential:**
test_get_replays
test_get_replays_started_at_sorted
test_get_replays_finished_at_sorted
test_get_replays_duration_sorted
test_get_replays_user_filters
test_get_replays_user_sorts

**Maybe:**
test_query_branches_string_conditions (device.brand)
test_query_branches_integer_conditions (count_errors)
test_query_branches_string_uuid_conditions (id)
test_query_branches_ip_address_conditions (user.ip_address)
test_query_branches_computed_activity_conditions (activity)
test_non_empty_string_scalar (dist)

test_query_invalid_ipv4_addresses
test_get_replays_user_filters_invalid_operator
test_get_replays_user_filters_invalid_value

**Comment by cmanallen** [comment]
You may need to rebuild your environment.  `sentry devservices rm snuba clickhouse && sentry devservices up`.

File: src/sentry/replays/usecases/query/__init__.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/replays/usecases/query/__init__.py:332]: No clue.  Drop it if its not useful to you.

File: src/sentry/replays/usecases/query/configs/materialized_view.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def activity_expression(count_errors, count_urls):
     "device_name": Function("anyIfMerge", parameters=[Column("device_name")]),
     "dist": Function("anyIfMerge", parameters=[Column("dist")]),
     "duration": Function("anyIfMerge", parameters=[Column("duration")]),
-    "finished_at": Function("maxMerge", parameters=[Column("finished_at")]),
+    "finished_at": Function("maxIfMerge", parameters=[Column("finished_at")]),
```
Comment (aliu39) [inline src/sentry/replays/usecases/query/configs/materialized_view.py:57]: Why do some fields use -IfMerge but others use -Merge? Don't we need to pass a condition as a parameter for -Ifs?

Comment (cmanallen) [reply src/sentry/replays/usecases/query/configs/materialized_view.py:57]: You only need the -If suffix if your aggregation function requires it.  Some fields do not have a conditional operator when grouping.  You can use the command `SHOW CREATE TABLE replays_aggregated_local` to see which fields require the -IfMerge suffix.

**Review by ryan953 — APPROVED** [review]

**Comment by cmanallen** [comment]
@ryan953 Has approved this.  All the back end engineers are on PTO.  This is a low risk change and I'm taking full responsibility for its deployment.

**Pull request merged**